lib/driver.js

```diff
@@ -411,11 +416,19 @@ class XCUITestDriver extends BaseDriver {
       await this.wda.quit();
       await this.wda.uninstall();
       this.logEvent('wdaUninstalled');
-    } else if (!util.hasValue(this.wda.webDriverAgentUrl) && (await this.wda.isRunning())) {
-      log.info(`Will reuse previously cached WDA instance at '${this.wda.url.href}'. ` +
-               `Set the wdaLocalPort capability to a value different from ${this.wda.url.port} ` +
-               `if this is an undesired behavior.`);
-      this.wda.webDriverAgentUrl = this.wda.url.href;
+    } else if (!util.hasValue(this.wda.webDriverAgentUrl)) {
+      const currentAppBundleID = (await this.wda.getRunningBundleId());
```

Comment:
I tried `[[NSBundle mainBundle] bundleIdentifier]` on various environment. `"bundleID" : "com.apple.test.WebDriverAgentRunner-Runner"` was the result of `[[NSBundle mainBundle] bundleIdentifier]` on Real Device (with personal account.) The bundle id wasn't it.
So, considering company/personal account, if we have `updatedWDABundleId`, then uninstall WDA is probably good.

```javascript
} else if (!util.hasValue(this.wda.webDriverAgentUrl)) {
    if (util.hasValue(this.opts.updatedWDABundleId)) {
        await this.wda.uninstall();
    } else if (await this.wda.isRunning())) {
        this.wda.webDriverAgentUrl = this.wda.url.href;
    }
}
```

Reply:
@KazuCocoa You could try to add 

```
      @"build" :
        @{
          @"time" : [self.class buildTimestamp],
          @"libBundleId": (id)[NSBundle bundleForClass:self].bundleIdentifier
        }
```

into `+ (id<FBResponsePayload>)handleGetStatus:(FBRouteRequest *)request` if this is the id you are looking for

Reply:
in my tests this returns

```
{
  "value" : {
    "state" : "success",
    "os" : {
      "name" : "iOS",
      "version" : "12.0",
      "sdkVersion" : "12.0"
    },
    "ios" : {
      "simulatorVersion" : "12.0",
      "ip" : "192.168.120.127"
    },
    "build" : {
      "time" : "Jun 28 2018 13:44:31",
      "libBundleId" : "com.facebook.wda.lib"
    }
  },
  "sessionId" : "152FDFFB-88D3-450E-ACEE-A41F6A421B77",
  "status" : 0
}
```

Reply:
Thanks! But I'm investigating is getting `PRODUCT_BUNDLE_IDENTIFIER` to identify it is same as `updatedWDABundleId` in Appium's capability.
(I could get the same result with https://github.com/appium/appium-xcuitest-driver/pull/709#discussion_r198811651 )